### PR TITLE
PAYARA-1551 created get-domain-xml command

### PR DESCRIPTION
**Pull request by Cousjava**

### CODE REVIEW

**Comment by Cousjava** [comment]
jenkins test please

**Comment by Cousjava** [comment]
jenkins test please

**Comment by Cousjava** [comment]
do not merge - some endpoint issues

File: appserver/payara-appserver-modules/zendesk-support/src/main/java/fish/payara/appserver/zendesk/config/DomainXMLConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.jvnet.hk2.config.ConfigBeanProxy;
+import org.jvnet.hk2.config.Configured;
+
+/**
+ *
+ * @author jonathan
+ */
+@Configured
+public interface DomainXMLConfiguration extends ConfigBeanProxy, ConfigExtension{
+
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/zendesk-support/src/main/java/fish/payara/appserver/zendesk/config/DomainXMLConfiguration.java]: Why have a config bean that doesn't store anything? Please remove

File: appserver/payara-appserver-modules/zendesk-support/src/main/java/fish/payara/appserver/zendesk/admin/GetDomainXMLCommand.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import org.glassfish.api.admin.ServerEnvironment;
+/**
+ *
+ * @author Jonathan Coustick
+ */
+@Service(name = "get-domain-xml")
+@PerLookup
+@ExecuteOn(RuntimeType.DAS)
+@RestEndpoints({
+    @RestEndpoint(configBean = DomainXMLConfiguration.class,
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/zendesk-support/src/main/java/fish/payara/appserver/zendesk/admin/GetDomainXMLCommand.java]: In line with my previous comment about removing the pointless configBean, this should instead exist under `ZendeskSupportConfiguration.class`

File: appserver/payara-appserver-modules/zendesk-support/pom.xml [hunk 3]
```diff
@@ -22,5 +22,17 @@
             <version>${project.version}</version>
             <type>jar</type>
         </dependency>
+        <dependency>
+            <groupId>org.glassfish.main.admin</groupId>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/zendesk-support/pom.xml]: Are either of these dependencies actually used anywhere? They don't seem to be

File: appserver/payara-appserver-modules/zendesk-support/src/main/java/fish/payara/appserver/zendesk/config/DomainXMLConfiguration.java (hunk 1 above)
Comment (Cousjava) [inline appserver/payara-appserver-modules/zendesk-support/src/main/java/fish/payara/appserver/zendesk/config/DomainXMLConfiguration.java]: A config bean is required to create a REST endpoint, but for the command there are no options for it and no way to configure it.

**Comment by Cousjava** [comment]
jenkins test please

**Comment by Cousjava** [comment]
endpoint works correctly now - management/domain/configs/config/server-config/zendesk-support-configuration/get-domain-xml
can be merged

File: appserver/payara-appserver-modules/zendesk-support/pom.xml [hunk 4]
```diff
@@ -22,5 +22,11 @@
             <version>${project.version}</version>
             <type>jar</type>
         </dependency>
+        <dependency>
+            <groupId>org.glassfish.main.common</groupId>
+            <artifactId>amx-core</artifactId>
+            <version>4.1.1.172-SNAPSHOT</version>
```
Comment (smillidge) [inline appserver/payara-appserver-modules/zendesk-support/pom.xml]: this should be ${project.version}
Also why is it required, I couldn't see anything in the service that needs it?

**Comment by Cousjava** [comment]
jenkins test please

**Pull request merged**